### PR TITLE
Animelib sync

### PR DESCRIPTION
**Pull request by DFreezerR**
Added sync to AnimeLib. Website upgraded it's UI and I decided to add it.  PR related to #2061 because it's a part of these websites. Had to add new player to `src/pages/playerUrls.js`  since it's their own player.
If there's someone who want to do it more profound way, say it and feel free to close it (telling this since original issue mentioned APIs and other websites in that group)

### CODE REVIEW

**Comment by DFreezerR** [comment]
need to fix selectors to api where I can since html classes are changing (not familiar with vue, I guess it changes every build?)

**Comment by DFreezerR** [comment]
Works now

**Comment by lolamtisch** [comment]
@DFreezerR 
Calling apis is always a bit of gamble, because of CORS etc.

In hidive we experimented with overriding the fetch of the website in proxy script and listening so on the requests the website does. No complain yet from the users. Do you think that could be an option too here?

https://github.com/MALSync/MALSync/blob/master/src/pages/Hidive/proxy.ts

**Comment by DFreezerR** [comment]
> @DFreezerR Calling apis is always a bit of gamble, because of CORS etc.
> 
> In hidive we experimented with overriding the fetch of the website in proxy script and listening so on the requests the website does. No complain yet from the users. Do you think that could be an option too here?
> 
> https://github.com/MALSync/MALSync/blob/master/src/pages/Hidive/proxy.ts

The main issue here (AnimeLib, MangaLib, RanobeLib) was that users can not access some title's chapters due to 2 options:
1. Title might be licensed so chapters/episodes got deleted - you can do nothing here
2. Title might be only read/watched if you auth into account - create account

Currently I made a logic getting title's info based on API, but if it fails, the code will try get info using selectors. So like a backup method in case user trying to read\watch something required an auth.

I think it's possible to do what you mean here, I just didn't see this option. Guess might add it like a 3rd variant just in case. Do you think overriding fetch would be better approach? I manly did API because several websites are using it + it was convenient.

**Comment by lolamtisch** [comment]
Then let's keep it like it is until it breaks with api

**Comment by DFreezerR** [comment]
> Then let's keep it like it is until it breaks with the api

Sure.
Even if API breaks, the selectors will back it up for some time.
The worst case scenario if API breaks:
1. AnimeLib - Can not get season number on sync page - optional
2. MangaLib - Can not get cover image on sync page - still optional / Can not get total number of chapters - also optional 
3. RanobeLib - same as p.2

Ideally yes, there should be something like you mentioned in case "some situations"
I'll look into this thing you linked and try implement it as "additional improvements", but a bit later

File: src/pages/AnimeLib/main.ts [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            return j.$('.tabs-item').length;
+          },
+          () => {
+            page.handlePage();
+          },
+          500,
+        );
+      }
+
+      page.handlePage();
```
Comment (lolamtisch) [inline src/pages/AnimeLib/main.ts:130]: This results in all overivew pages beeing triggered 2 times. I imagine you wanted inside of  https://github.com/MALSync/MALSync/pull/2606/files#diff-ad21d95b5e6745de2f7f26e52df7f79efe09ffce69150b485a72cb64e069ca45R102

File: src/pages/playerUrls.js [hunk 2]
```diff
@@ -741,4 +741,8 @@ module.exports = {
   doflix: {
     match: ['*://doflix.net/*'],
   },
+  // anilib
+  anilib: {
```
Comment (lolamtisch) [inline src/pages/playerUrls.js:745]: I did not find any player that would need this. Do you have an example. If the player is directly on the page without an iframe then no additional injection is necesseary.

File: src/pages/AnimeLib/main.ts (hunk 1 above)
Comment (DFreezerR) [reply src/pages/AnimeLib/main.ts:130]: Ah, yes, this need to be in sync one

File: src/pages/playerUrls.js (hunk 2 above)
Comment (DFreezerR) [reply src/pages/playerUrls.js:745]: Was long time since then, was thought that any new player should be added manually, fixed it

**Review by lolamtisch — APPROVED** [review]

**Pull request merged**